### PR TITLE
DB Access Layer Merges: GetTablePattern ...

### PR DESCRIPTION
**Pull request by a-barboza**
Added the following APIs:

ExistKeysPattern()
GetTablePattern()
Key.IsAllKeyPattern()
Go UT, and Benchmark Tests for the same
TranslibDBScriptFail error type to translib/tlerr package

### CODE REVIEW

**Review by kwangsuk — APPROVED** [review]

File: translib/db/db_table_pattern_test.go [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	if e != nil {
+		t.Errorf("GetTable() returns err: %v", e)
+	}
+
+	// The ordering of the arrays in patterns map values matters
+	delete(tPat.patterns, db.key2redis(&ts, Key{Comp: []string{"*"}}))
+	delete(tOrig.patterns, db.key2redis(&ts, Key{Comp: []string{"*"}}))
+
+	if !reflect.DeepEqual(tPat, tOrig) {
+		fmt.Println("\ntPat: \n", tPat)
```
Comment (sallylsy) [inline translib/db/db_table_pattern_test.go:100]: Why not t.Logf()

File: translib/db/db_table_pattern_test.go [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+		t.Errorf("GetTablePattern() returns err: %v", e)
+	}
+
+	tOrig, e := db.GetTable(&tsEmpty)
+	if e != nil {
+		t.Errorf("GetTable() returns err: %v", e)
+	}
+
+	if !reflect.DeepEqual(tPat, tOrig) {
+		fmt.Println("\ntPat: \n", tPat)
```
Comment (sallylsy) [inline translib/db/db_table_pattern_test.go:119]: Same. Suggest to replace fmt.Println() to  t.Logf()

File: translib/db/db_keys_pattern.go [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+	// Register the Lua Script
+	luaScriptExistsKeysPatterns = redis.NewScript(`
+		for i,k in pairs(redis.call('KEYS', KEYS[1])) do
+			return 'true'
+		end
+		return 'false'
+	`)
+
+	// Alternate Lua Script
+	// luaScriptExistsKeysPatterns = redis.NewScript(`
```
[the remark this answers is not in the document]
Comment (a-barboza) [reply translib/db/db_keys_pattern.go:192]: It is a comment, to indicate an alternate implementation. I would rather keep it.

File: translib/db/db_table_pattern.go [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+
+	// Check Per Connection Cache first.
+	if (d.dbCacheConfig.PerConnection &&
+		d.dbCacheConfig.isCacheTable(ts.Name)) ||
+		(d.Opts.IsOnChangeEnabled && d.onCReg.isCacheTable(ts.Name)) {
+
+		if cTable, ok := d.cache.Tables[ts.Name]; ok && cTable.complete {
+
+			// Copy relevent keys of cTable to table, and set keys
+			// Be aware of cache poisoning
```
[the remark this answers is not in the document]
Comment (a-barboza) [reply translib/db/db_table_pattern.go:107]: Yes, it is handled by the Copy() on L113

**Review by sallylsy — APPROVED** [review]

File: translib/db/db_table_pattern_test.go (hunk 2 above)
Comment (a-barboza) [reply translib/db/db_table_pattern_test.go:119]: A review has been raised for fixing the issues raised in the review comments: 
https://github.com/sonic-net/sonic-mgmt-common/pull/106

**Pull request merged**